modin/experimental/xgboost/test/test_dmatrix.py

```diff
@@ -0,0 +1,174 @@
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+
+import pytest
+
+import modin.experimental.xgboost as mxgb
+import modin.pandas as pd
+import numpy as np
+
+import multiprocessing as mp
+import ray
+
+import xgboost as xgb
+
+
+ray.init(log_to_driver=False)
+
+num_cpus = mp.cpu_count()
+
+
+rng = np.random.RandomState(1994)
+
+
+class TestDMatrix:
+    def test_dmatrix_numpy_init(self):
+        data = np.random.randn(5, 5)
+        dm = xgb.DMatrix(data)
+        md_data = pd.DataFrame(data)
+        md_dm = mxgb.DMatrix(md_data)
+        assert md_dm.num_row() == dm.num_row() == 5
+        assert md_dm.num_col() == dm.num_col() == 5
+
+        data = np.array([[1, 2], [3, 4]])
+        dm = xgb.DMatrix(data)
+        md_data = pd.DataFrame(data)
+        md_dm = mxgb.DMatrix(md_data)
+        assert md_dm.num_row() == dm.num_row() == 2
+        assert md_dm.num_col() == dm.num_col() == 2
+
+        # 0d array
+        with pytest.raises(ValueError):
+            xgb.DMatrix(np.array(1))
+        with pytest.raises(ValueError):
+            mxgb.DMatrix(pd.DataFrame(np.array(1)))
+        # 3d array
+        data = np.random.randn(5, 5, 5)
+        with pytest.raises(ValueError):
+            xgb.DMatrix(data)
+        with pytest.raises(ValueError):
+            mxgb.DMatrix(pd.DataFrame(data))
+        # object dtype
+        data = np.array([["a", "b"], ["c", "d"]])
+        with pytest.raises(ValueError):
+            xgb.DMatrix(data)
+        with pytest.raises(ValueError):
+            mxgb.DMatrix(pd.DataFrame(data))
+
+    def test_feature_names_slice(self):
+        data = np.random.randn(5, 5)
+
+        # different length
+        with pytest.raises(ValueError):
+            xgb.DMatrix(data, feature_names=list("abcdef"))
+        with pytest.raises(ValueError):
+            mxgb.DMatrix(pd.DataFrame(data), feature_names=list("abcdef"))
+
+        # contains duplicates
+        with pytest.raises(ValueError):
+            xgb.DMatrix(data, feature_names=["a", "b", "c", "d", "d"])
+        with pytest.raises(ValueError):
+            mxgb.DMatrix(pd.DataFrame(data), feature_names=["a", "b", "c", "d", "d"])
+
+        # contains symbol
+        with pytest.raises(ValueError):
+            xgb.DMatrix(data, feature_names=["a", "b", "c", "d", "e<1"])
+        with pytest.raises(ValueError):
+            mxgb.DMatrix(pd.DataFrame(data), feature_names=["a", "b", "c", "d", "e<1"])
+
```

Comment:
This code might be unified. We can create a separate func, which could check equality of DMatrix. Possible example is https://github.com/modin-project/modin/blob/0a642753af2cd2332ac71166bcddc89b0cace9d5/modin/pandas/test/utils.py#L662

Reply:
function was unified.